mlx_graphs/datasets/dataset.py

```diff
@@ -166,11 +166,20 @@ def _load(self):
             self._save()
 
     def _num_classes(self, task: Literal["node", "edge", "graph"]) -> int:
-        flattened_labels = [
-            getattr(g, f"{task}_labels")
-            for g in self.graphs
-            if getattr(g, f"{task}_labels") is not None
-        ]
+        flattened_labels = []
+        for g in self.graphs:
+            if isinstance(g, GraphData):
+                labels = getattr(g, f"{task}_labels")
+                if labels is not None:
+                    flattened_labels.append(labels)
+            elif isinstance(g, HeteroGraphData):
+                if task == "node":
+                    labels = [v for v in g.collect(f"{task}_labels", True).values()]
+                else:
+                    labels = [getattr(g, f"{task}_labels", None)]
+                flattened_labels.extend(
+                    [label for label in labels if label is not None]
+                )
```

Comment:
Would it make more sense to return a dict with the classes per node/edge type?

Reply:
Wouldn't this also change the way we handle `GraphData`?

Reply:
yes, that's why I was a bit concerned about having the same dataset class for `GraphData` and `HeteroGraphData`. I guess we can keep it as is for now and change it later if needed.